plugins/module_utils/prism/clusters.py

```diff
@@ -5,13 +5,104 @@
 __metaclass__ = type
 
 from .prism import Prism
+from copy import deepcopy
 
 
 class Cluster(Prism):
     def __init__(self, module):
         resource_type = "/clusters"
         super(Cluster, self).__init__(module, resource_type=resource_type)
+        self.build_spec_methods = {
+            "authorized_public_key_list": self._build_spec_public_keys,
+            "timezone": self._build_spec_timezone,
+            "supported_information_verbosity": self._build_spec_supported_information_verbosity,
+            "redundancy_factor": self._build_spec_redundancy_factor,
+            "network": self._build_spec_network,
+        }
 
+    def _get_default_spec(self):
+        return deepcopy(
+            {
+                "api_version": "3.1.0",
+                "metadata": {"kind": "cluster", "spec_version": 2},
+                "spec": {
+                    "name": None,
+                    "resources": {
+                        "config": {
+                            "authorized_public_key_list": []
+                        }
+                    }
+                }
+            }
+        )
+
+    def _build_spec_redundancy_factor(self, payload, redundancy_factor):
```

Comment:
if redundancy_factor is none from user then it would never reach here. As None values are omitted in Entity.get_spec() and even from ansible module.params. This applies for all fields from input so we can change accordingly in all build spec methods

Reply:
Although we can restrict it from passing 0